### ISSUE TITLE
Recipe Compile Error in /tmp/chef/cookbooks/postfix/recipes/sasl_auth.rb

### ISSUE THREAD

**Issue by CloCkWeRX** [chef-cookbooks/postfix#76]
By default, the recipe doesn't work unless you override attributes to enable `smtp_sasl_auth_enable => 'yes'`

```
Recipe Compile Error in /tmp/chef/cookbooks/postfix/recipes/sasl_auth.rb
================================================================================


ArgumentError
-------------
You must supply a name when declaring a template resource


Cookbook Trace:
---------------
  /tmp/chef/cookbooks/postfix/recipes/sasl_auth.rb:51:in `from_file'


Relevant File Content:
----------------------
/tmp/chef/cookbooks/postfix/recipes/sasl_auth.rb:

 44:  
 45:  execute 'postmap-sasl_passwd' do
 46:    command "postmap #{node['postfix']['sasl_password_file']}"
 47:    environment 'PATH' => "#{ENV['PATH']}:/opt/omni/bin:/opt/omni/sbin" if platform_family?('omnios')
 48:    action :nothing
 49:  end
 50:  
 51>> template node['postfix']['sasl_password_file'] do
 52:    source 'sasl_passwd.erb'
 53:    owner 'root'
 54:    group 'root'
 55:    mode 0400
 56:    notifies :run, 'execute[postmap-sasl_passwd]', :immediately
 57:    notifies :restart, 'service[postfix]'
 58:    variables(settings: node['postfix']['sasl'])
 59:  end
 60:  
```

This is likely due to the conditional configuration of `sasl_password_file`

```
if node['postfix']['main']['smtp_sasl_auth_enable'] == 'yes'
  default['postfix']['sasl_password_file'] = "#{node['postfix']['conf_dir']}/sasl_passwd"
  default['postfix']['main']['smtp_sasl_password_maps'] = "hash:#{node['postfix']['sasl_password_file']}"
  default['postfix']['main']['smtp_sasl_security_options'] = 'noanonymous'
  default['postfix']['sasl']['smtp_sasl_user_name'] = ''
  default['postfix']['sasl']['smtp_sasl_passwd']    = ''
  default['postfix']['main']['relayhost'] = ''
end
```

**Comment by chasebolt** [comment]
ive done several testing and this issue looks to be resolved by commit 489bfaa, @CloCkWeRX please close this.

**Comment by tas50** [comment]
Closing this out since @chasebolt reports it has been fixed
